Public/Src/Engine/Processes/AsyncProcessExecutor.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Diagnostics.ContractsLight;
+using System.Threading.Tasks;
+using BuildXL.Interop;
+using BuildXL.Native.IO;
+using BuildXL.Utilities;
+using BuildXL.Utilities.Tasks;
+
+namespace BuildXL.Processes
+{
+    /// <summary>
+    /// Executes process asynchronously.
+    /// </summary>
+    public class AsyncProcessExecutor : IDisposable

```

Comment:
👍

Reply:
Thanks for the thumb. How did you produce it?

---
In reply to: [277335432](https://github.com/Microsoft/BuildXL/pull/159#discussion_r277335432) [](ancestors = 277335432)

Reply:
from GitHub UI: I just press type `:` and wait for an automatic completion to offer the thumbs up emoji